### ISSUE TITLE
[Bash] Wrong highlighting for inline comment in backticks

### ISSUE THREAD

**Issue by jfcherng** [sublimehq/Packages#2408]
```bash

sed -i'' \
    `# comment 1` \
    -e 's/search 1/replace 1/g' \
    `: # comment 2` \
    -e 's/search 2/replace 2/g' \
    "input.txt"
```

Both `` `# comment 1` \ `` and `` `: # comment 2` \ `` are wrong.

**Expected** (Photoshop-ed):
![image](https://user-images.githubusercontent.com/6594915/87517661-bff16200-c6b1-11ea-9bd4-507a51cd9a4d.png)

**Actual**:
![image](https://user-images.githubusercontent.com/6594915/87517337-4d808200-c6b1-11ea-87d9-59e02d881bbe.png)

- Sublime Version: 4077
- OS Version: Win7 x64

**Comment by deathaxe** [comment]
This issue is highly related with https://github.com/sublimehq/Packages/issues/1781

I still see this issue on my https://github.com/deathaxe/sublime-packages/tree/pr/shellscript/patchset-2020 branch which includes a fix for #1781.

With regards to how comments are implemented I tend to think it to be a syntax engine bug.

https://github.com/sublimehq/Packages/blob/03babb588a03ce5a1d9fd3592ce7bc265ee90d49/ShellScript/Bash.sublime-syntax#L68-L76

As the content within the backticks is `embed`ed, I'd expect `^\s*` to match directly after the opening backtick, because from the perspective of the embedded syntax the first line starts right after it.

**Comment by wbond** [comment]
> As the content within the backticks is `embed`ed, I'd expect `^\s*` to match directly after the opening backtick, because from the perspective of the embedded syntax the first line starts right after it.

Interesting observation. Currently I don't think we set the flags for this to be true, but I'd have to look into it.

**Comment by jfcherng** [comment]
Note that the highlighting for `` `: # comment 2` \ `` is wrong as well.

**Comment by deathaxe** [comment]
Currently, yes. But without the other issue this one would just be a duplicate.

**Comment by jfcherng** [comment]
So I guess I can close this?

**Comment by deathaxe** [comment]
Maybe should keep it open for a while due to the linestart matching in embedded syntaxes issue or raise it at the coreissue site.

**Comment by deathaxe** [comment]
I was able to remove the `(?:^\s*|\s+)` from comments match. So this issue should be resolved with the pending PR with or without core changes with regards to handling `^` at the beginning of embedded syntaxes.

**Comment by jfcherng** [comment]
Maybe add `Fixes #2408` in the PR :D
